model-optimizer/extensions/back/InterpolateReshape.py

```diff
@@ -82,14 +83,18 @@ def make_interpolate_reshapeable(interpolate, concat):
 
         shape = Shape(graph, {'name': src.node.soft_get('name', src.node.id) + '/Shape'}).create_node()
         shape.in_port(0).connect(src)
-        gather = create_op_with_const_inputs(graph, Gather, {1: np.array(interpolate.axes, dtype=np.int32), 2: int64_array(0)},
+        gather = create_op_with_const_inputs(graph, Gather,
+                                             {1: np.array(interp_axes, dtype=np.int32), 2: int64_array(0)},
                                              {'name': shape.name + '/Gathered'}, shape)
         interpolate.in_port(1).get_connection().set_source(gather.out_port(0))
 
     def find_and_replace_pattern(self, graph: Graph):
         for interpolate in graph.get_op_nodes(type='Interpolate'):
+            num_inputs = len([p for p in interpolate.in_ports().values() if not p.disconnected()])
             if interpolate.in_port(1).get_source().node.soft_get('type') != 'Const':
                 continue
+            if num_inputs == 3 and interpolate.in_port(2).get_source().node.soft_get('type') != 'Const':
```

Comment:
I would rather check the 'version' of the Interpolate in the `graph.get_op_nodes(type='Interpolate'):` in the `find_and_replace_pattern ` and do not apply the transformation for the Interpolate-4.

@jane-intel, could we skip this transformation for Interpolate-4?

Reply:
Fixed.